### DISCUSSION TITLE
Using the master scripts to configure prereqs for runner on our VM

### DISCUSSION THREAD

**Discussion by prmishra-athoc** [actions/virtual-environments#3843] [Ideas]
As per my understanding the json file and scripts are used to create an image and which gets deployed for the runner. Do we have a master script which reads the json file and deploys it on our own VM. I went through the repo but found some scripts to install tools and softwares but could not locate any master script which uses the .json script content and deploys prereqs for the runner.

**Comment by miketimofeev** [comment]
HI @prmishra-athoc!
Please take a look at this guide, it contains links to the scripts for generation and vm creation
https://github.com/actions/virtual-environments/blob/main/docs/create-image-and-azure-resources.md

**Reply by prmishra-athoc** [reply]
Hi @miketimofeev 
I did take a look at this which creates an image and that image is used to deploy on to the azure VM which configures and creates a hosted runner. However in my case we have our own VM(2016 and 2019 ready) and wanted to spin it up or install with the prereqs software for latest linux\windows runner and install the hosted runner on it.

**Reply by miketimofeev** [reply]
in that case, you need to choose another builder in packer templates like Hyper-V. We don't have much experience in such configurations, unfortunatel, but you can find a few pieces of advice here https://github.com/actions/virtual-environments/issues/931
